### ISSUE TITLE
Forced loop

### ISSUE THREAD

**Issue by west1778** [dev7355608/perfect-vision#53]
With the newest version of PV installed was causing Foundry to force a loop action no errors was thrown that I could see though but once I rolled it back it stopped happening. Sorry I don't have more info I'm sure someone else will report it with more info though.

**Comment by otigon** [comment]
Conflicting with MIDI-qol

**Comment by Hebelios** [comment]
It's probably stuck in some kind of scene-update loop I'd guess?
One thing checks an option where you have to reload the game for, the other does the same.

**Comment by linden-dg** [comment]
[Posted this on Midi-QoL as well - [https://gitlab.com/tposney/midi-qol/-/issues/216](https://gitlab.com/tposney/midi-qol/-/issues/216) ]

When you have the Perfect Vision module and Midi-QoL (https://gitlab.com/tposney/midi-qol) active at the same time, Foundry goes into an infinite refresh loop.
It's a really nasty bug because Foundry won't even load up enough to get into the Module management to unselect them. The only way to fix it is to completely uninstall either module - took me about an hour of setting up blank worlds and activating modules one-by-one to find it.

System details:
- Foundry `0.7.9`
- DnD5e `1.2.2`
- Midi QOL `0.3.59`
- Perfect Vision `1.7.0`

Conflict/issue happens on a completely fresh world with only those two modules active. This only happens when the two of them are both active - either by themselves are fine. 

Not sure if it's helpful, but attached full console log at the end. In particular, the last line is triggering a refresh of Foundry.

![image](https://user-images.githubusercontent.com/5335330/105570299-5b743580-5dad-11eb-97e4-de4fa4b31eb7.png)


```
_______________________________________________________________
 _____ ___  _   _ _   _ ____  ______   __ __     _______ _____ 
|  ___/ _ \| | | | \ | |  _ \|  _ \ \ / / \ \   / |_   _|_   _|
| |_ | | | | | | |  \| | | | | |_) \ V /   \ \ / /  | |   | |  
|  _|| |_| | |_| | |\  | |_| |  _ < | |     \ V /   | |   | |  
|_|   \___/ \___/|_| \_|____/|_| \_\|_|      \_/    |_|   |_|  
===============================================================
foundry.js:6348 Foundry VTT | Reestablishing existing session nap7i1umqlsqrsmnc9he5bhe
content_script.js:71 Rivet content_script initialised
foundry.js:6386 Foundry VTT | Connected to server socket using session nap7i1umqlsqrsmnc9he5bhe
foundry.js:6421 Foundry VTT | The game World is currently active
foundry.js:6455 Foundry VTT | Initializing Game instance
systems/dnd5e/dnd5e.js:46 DnD5e | Initializing the DnD5e Game System
_______________________________
______      ______ _____ _____
|  _  \___  |  _  \  ___|  ___|
| | | ( _ ) | | | |___ \| |__
| | | / _ \/\ | | |   \ \  __|
| |/ / (_>  < |/ //\__/ / |___
|___/ \___/\/___/ \____/\____/
_______________________________
modules/midi-qol/midi-qol.js:59 midi-qol | Initializing midi-qol
modules/perfect-vision/scripts/patch.js:4 Perfect Vision | Patching SightLayer.prototype.refresh (PRE)
modules/perfect-vision/scripts/patch.js:4 Perfect Vision | Patching PointSource.prototype.drawLight (POST)
modules/perfect-vision/scripts/patch.js:4 Perfect Vision | Patching PointSource.prototype._createContainer (POST)
modules/perfect-vision/scripts/patch.js:4 Perfect Vision | Patching PointSource.prototype.initialize (WRAPPER)
modules/perfect-vision/scripts/patch.js:4 Perfect Vision | Patching PointSource.prototype._initializeBlending (POST)
modules/perfect-vision/scripts/patch.js:4 Perfect Vision | Patching PointSource.prototype.drawLight (WRAPPER)
modules/perfect-vision/scripts/patch.js:4 Perfect Vision | Patching LightingLayer.prototype.draw (POST)
modules/perfect-vision/scripts/patch.js:4 Perfect Vision | Patching LightingLayer.prototype._configureChannels (WRAPPER)
modules/perfect-vision/scripts/patch.js:4 Perfect Vision | Patching LightingLayer.prototype._drawIlluminationContainer (POST)
modules/perfect-vision/scripts/patch.js:4 Perfect Vision | Patching LightingLayer.prototype.refresh (WRAPPER)
modules/perfect-vision/scripts/patch.js:4 Perfect Vision | Patching Token.prototype.updateSource (PRE)
modules/perfect-vision/scripts/patch.js:4 Perfect Vision | Patching TemplateLayer.prototype.activate (POST)
modules/perfect-vision/scripts/patch.js:4 Perfect Vision | Patching TemplateLayer.prototype.deactivate (POST)
modules/perfect-vision/scripts/patch.js:4 Perfect Vision | Patching BackgroundLayer.prototype.draw (POST)
modules/perfect-vision/scripts/patch.js:4 Perfect Vision | Patching EffectsLayer.prototype.draw (POST)
modules/perfect-vision/scripts/patch.js:4 Perfect Vision | Patching Token.prototype.draw (POST)
modules/perfect-vision/scripts/patch.js:4 Perfect Vision | Patching Tile.prototype.draw (POST)
modules/perfect-vision/scripts/patch.js:4 Perfect Vision | Patching MeasuredTemplate.prototype.draw (POST)
modules/perfect-vision/scripts/patch.js:4 Perfect Vision | Patching SightLayer.prototype.draw (POST)
modules/perfect-vision/scripts/patch.js:4 Perfect Vision | Patching SightLayer.prototype.tearDown (PRE)
modules/perfect-vision/scripts/patch.js:4 Perfect Vision | Patching Canvas.prototype._updateBlur (POST)
modules/perfect-vision/scripts/patch.js:4 Perfect Vision | Patching PointSource.prototype.drawLight (PRE)
modules/perfect-vision/scripts/patch.js:4 Perfect Vision | Patching LightingLayer.prototype.refresh (PRE)
modules/perfect-vision/scripts/patch.js:4 Perfect Vision | Patching BackgroundLayer.prototype.draw (POST)
modules/perfect-vision/scripts/patch.js:4 Perfect Vision | Patching EffectsLayer.prototype.draw (POST)
modules/perfect-vision/scripts/patch.js:4 Perfect Vision | Patching EffectsLayer.layerOptions (POST)
modules/perfect-vision/scripts/patch.js:4 Perfect Vision | Patching LightingLayer.prototype._configureChannels (POST)
foundry.js:3983 Foundry VTT | Synchronized official game time in 15ms
foundry.js:5956 Foundry VTT | Retrieved and compiled template systems/dnd5e/templates/actors/parts/active-effects.html
foundry.js:5956 Foundry VTT | Retrieved and compiled template systems/dnd5e/templates/actors/parts/actor-traits.html
foundry.js:5956 Foundry VTT | Retrieved and compiled template systems/dnd5e/templates/actors/parts/actor-inventory.html
foundry.js:5956 Foundry VTT | Retrieved and compiled template systems/dnd5e/templates/actors/parts/actor-features.html
foundry.js:5956 Foundry VTT | Retrieved and compiled template systems/dnd5e/templates/actors/parts/actor-spellbook.html
foundry.js:5956 Foundry VTT | Retrieved and compiled template systems/dnd5e/templates/items/parts/item-action.html
foundry.js:5956 Foundry VTT | Retrieved and compiled template systems/dnd5e/templates/items/parts/item-activation.html
foundry.js:5864 Foundry VTT | Loaded localization file lang/en.json
foundry.js:5864 Foundry VTT | Loaded localization file modules/midi-qol/lang/en.json
foundry.js:5864 Foundry VTT | Loaded localization file systems/dnd5e/lang/en.json
modules/midi-qol/midi-qol.js:28 midi-qol |  Patching  itemAttack {methodName: "rollAttack", target: "CONFIG.Item.entityClass.prototype.rollAttack", roll: ƒ, class: ƒ, replacement: ƒ}
modules/midi-qol/midi-qol.js:28 midi-qol |  Patching  itemDamage {methodName: "rollDamage", target: "CONFIG.Item.entityClass.prototype.rollDamage", roll: ƒ, class: ƒ, replacement: ƒ}
modules/midi-qol/midi-qol.js:28 midi-qol |  Patching  itemRoll {methodName: "roll", target: "CONFIG.Item.entityClass.prototype.roll", roll: ƒ, class: ƒ, replacement: ƒ}
modules/midi-qol/midi-qol.js:28 midi-qol |  Vision patching -  disabled
modules/midi-qol/module/setupModules.js:20 module about-time not active - some features disabled
setupModules @ modules/midi-qol/module/setupModules.js:20
(anonymous) @ modules/midi-qol/midi-qol.js:78
_call @ foundry.js:2496
callAll @ foundry.js:2456
setupGame @ foundry.js:6536
_initializeGameView @ foundry.js:7319
initialize @ foundry.js:6471
async function (async)
initialize @ foundry.js:6463
window.addEventListener.once @ foundry.js:7449
async function (async)
window.addEventListener.once @ foundry.js:7448
modules/midi-qol/module/setupModules.js:20 module dice-so-nice not active - some features disabled
setupModules @ modules/midi-qol/module/setupModules.js:20
(anonymous) @ modules/midi-qol/midi-qol.js:78
_call @ foundry.js:2496
callAll @ foundry.js:2456
setupGame @ foundry.js:6536
_initializeGameView @ foundry.js:7319
initialize @ foundry.js:6471
async function (async)
initialize @ foundry.js:6463
window.addEventListener.once @ foundry.js:7449
async function (async)
window.addEventListener.once @ foundry.js:7448
modules/midi-qol/module/setupModules.js:20 module lmrtfy not active - some features disabled
setupModules @ modules/midi-qol/module/setupModules.js:20
(anonymous) @ modules/midi-qol/midi-qol.js:78
_call @ foundry.js:2496
callAll @ foundry.js:2456
setupGame @ foundry.js:6536
_initializeGameView @ foundry.js:7319
initialize @ foundry.js:6471
async function (async)
initialize @ foundry.js:6463
window.addEventListener.once @ foundry.js:7449
async function (async)
window.addEventListener.once @ foundry.js:7448
modules/midi-qol/module/setupModules.js:20 module dae not active - some features disabled
setupModules @ modules/midi-qol/module/setupModules.js:20
(anonymous) @ modules/midi-qol/midi-qol.js:78
_call @ foundry.js:2496
callAll @ foundry.js:2456
setupGame @ foundry.js:6536
_initializeGameView @ foundry.js:7319
initialize @ foundry.js:6471
async function (async)
initialize @ foundry.js:6463
window.addEventListener.once @ foundry.js:7449
async function (async)
window.addEventListener.once @ foundry.js:7448
foundry.js:4503 Foundry VTT | Rendering SceneNavigation
foundry.js:4503 Foundry VTT | Rendering Notifications
foundry.js:4503 Foundry VTT | Rendering Sidebar
foundry.js:4503 Foundry VTT | Rendering PlayerList
foundry.js:4503 Foundry VTT | Rendering Hotbar
foundry.js:4503 Foundry VTT | Rendering Pause
foundry.js:4503 Foundry VTT | Rendering SceneControls
foundry.js:5956 Foundry VTT | Retrieved and compiled template systems/dnd5e/templates/items/parts/item-description.html
foundry.js:5956 Foundry VTT | Retrieved and compiled template systems/dnd5e/templates/items/parts/item-mountable.html
foundry.js:5956 Foundry VTT | Retrieved and compiled template modules/midi-qol/templates/saves.html
foundry.js:5956 Foundry VTT | Retrieved and compiled template modules/midi-qol/templates/hits.html
foundry.js:5956 Foundry VTT | Retrieved and compiled template modules/midi-qol/templates/item-card.html
foundry.js:5956 Foundry VTT | Retrieved and compiled template modules/midi-qol/templates/tool-card.html
foundry.js:5956 Foundry VTT | Retrieved and compiled template modules/midi-qol/templates/config.html
foundry.js:5956 Foundry VTT | Retrieved and compiled template modules/midi-qol/templates/damage-results.html
foundry.js:5956 Foundry VTT | Retrieved and compiled template templates/hud/navigation.html
foundry.js:5956 Foundry VTT | Retrieved and compiled template templates/sidebar/sidebar.html
foundry.js:4503 Foundry VTT | Rendering ChatLog
foundry.js:5956 Foundry VTT | Retrieved and compiled template templates/user/players.html
foundry.js:5956 Foundry VTT | Retrieved and compiled template templates/hud/hotbar.html
foundry.js:5956 Foundry VTT | Retrieved and compiled template templates/hud/pause.html
foundry.js:5956 Foundry VTT | Retrieved and compiled template templates/hud/controls.html
foundry.js:5956 Foundry VTT | Retrieved and compiled template templates/sidebar/chat-log.html
foundry.js:4503 Foundry VTT | Rendering CombatTracker
foundry.js:5956 Foundry VTT | Retrieved and compiled template templates/sidebar/combat-tracker.html
foundry.js:4503 Foundry VTT | Rendering ActorDirectory
foundry.js:5956 Foundry VTT | Retrieved and compiled template templates/sidebar/actor-directory.html
foundry.js:4503 Foundry VTT | Rendering ItemDirectory
foundry.js:6688 Foundry VTT | Document fonts loaded and ready
pixi.min.js:8 
    PixiJS 5.3.4 - ✰ WebGL 2 ✰      http://www.pixijs.com/    ♥♥♥ 
foundry.js:10151 Foundry VTT | Skipping game canvas - no active scene.
modules/midi-qol/midi-qol.js:28 midi-qol |  Patching rollAbilitySave
modules/midi-qol/midi-qol.js:28 midi-qol |  Patching rollAbilityTest
modules/midi-qol/midi-qol.js:28 midi-qol |  Patching rollSkill
modules/midi-qol/midi-qol.js:28 midi-qol |  Patching rollDeathSave
foundry.js:5956 Foundry VTT | Retrieved and compiled template templates/sidebar/item-directory.html
foundry.js:4503 Foundry VTT | Rendering JournalDirectory
foundry.js:5956 Foundry VTT | Retrieved and compiled template templates/sidebar/journal-directory.html
foundry.js:4503 Foundry VTT | Rendering RollTableDirectory
foundry.js:5956 Foundry VTT | Retrieved and compiled template templates/sidebar/roll-table-directory.html
foundry.js:4503 Foundry VTT | Rendering PlaylistDirectory
foundry.js:5956 Foundry VTT | Retrieved and compiled template templates/sidebar/playlist-directory.html
foundry.js:4503 Foundry VTT | Rendering CompendiumDirectory
foundry.js:5956 Foundry VTT | Retrieved and compiled template templates/sidebar/compendium.html
foundry.js:4503 Foundry VTT | Rendering Settings
modules/perfect-vision/scripts/extend.js:5 Uncaught (in promise) TypeError: Invalid value used as weak map key
    at WeakMap.set (<anonymous>)
    at extend (modules/perfect-vision/scripts/extend.js:5)
    at LightingLayer.<anonymous> (modules/perfect-vision/scripts/vision.js:611)
    at LightingLayer.wrapper (modules/perfect-vision/scripts/patch.js:80)
    at LightingLayer.wrapper (modules/perfect-vision/scripts/patch.js:72)
    at refresh (modules/perfect-vision/scripts/migrate.js:302)
    at modules/perfect-vision/scripts/migrate.js:397
extend @ modules/perfect-vision/scripts/extend.js:5
(anonymous) @ modules/perfect-vision/scripts/vision.js:611
wrapper @ modules/perfect-vision/scripts/patch.js:80
wrapper @ modules/perfect-vision/scripts/patch.js:72
refresh @ modules/perfect-vision/scripts/migrate.js:302
(anonymous) @ modules/perfect-vision/scripts/migrate.js:397
async function (async)
(anonymous) @ modules/perfect-vision/scripts/migrate.js:393
_call @ foundry.js:2496
callAll @ foundry.js:2456
setupGame @ foundry.js:6560
async function (async)
setupGame @ foundry.js:6539
_initializeGameView @ foundry.js:7319
initialize @ foundry.js:6471
async function (async)
initialize @ foundry.js:6463
window.addEventListener.once @ foundry.js:7449
async function (async)
window.addEventListener.once @ foundry.js:7448
Navigated to https://foundry.xxx.app/game
```

**Comment by MS-PBS** [comment]
I can reproduce this. Rolling back to the previous version of Perfect Vision (1.6.2) fixed it for me.

**Comment by dev7355608** [comment]
Thanks for the report. I released an emergency update. The code that's causing the problem is currently not necessary anyway fortunately, so I disabled it for now. I'll need to investigate later when I have the time.

**Comment by dev7355608** [comment]
Found the source of the problem and fixed it in v1.7.2.

Unfortunately this bug caused the Midi QOL setting *Players control owned hidden tokens* to be reset to its default (off).
